src/content/docs/en/recipes/minification.mdx

```diff
@@ -0,0 +1,61 @@
+---
+title: Minify HTML responses with middleware
+description: Use Astro middleware to minify HTML responses
+i18nReady: false
+type: recipe
+---
+
+import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'
+
+Astro supports built-in [HTML minification](/en/reference/configuration-reference/#compresshtml) which works for `.astro` components, but not for [islands](/en/concepts/islands/). This recipe shows how to use Astro middleware to minify complete HTML responses.
+
+## Recipe
+
+1. Install the `html-minifier` package:
+
+  <PackageManagerTabs>
+    <Fragment slot="npm">
+    ```shell
+    npm install html-minifier
+    ```
+    </Fragment>
+    <Fragment slot="pnpm">
+    ```shell
+    pnpm install html-minifier
+    ```
+    </Fragment>
+    <Fragment slot="yarn">
+    ```shell
+    yarn add html-minifier
+    ```
+    </Fragment>
+  </PackageManagerTabs>
+
+2. Create a middleware file `src/middleware.ts`:
+
+  ```ts
+  import {minify} from "html-minifier";
+
+  export const onRequest = async (context, next) => {
+    const response = await next();
+
+    // return early if the response is not a HTML response
+    if (
+      !response.text ||
+      !response.headers.get('content-type')?.includes('text/html')
+    ) return new Response(response.body, {status: response.status, headers: response.headers})
+
+    const html = await response.text();
+    const minifiedHtml = minify(html)
+    
+    return new Response(minifiedHtml, {
+      status: 200,
+      headers: response.headers
+    });
+  }
+  ```
+
```

Comment:
```suggestion

4. Check / Inspect using dev tools...
```
I like some kind of guidance here, even if it's not to compare from before, but give them something they can look at. Can they see the minified HTML somewhere? Can they see a smaller bundle? Give them some kind of conclusion here!